datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java

```diff
@@ -79,6 +126,107 @@ public static GraphQLEngine get() {
         return _engine;
     }
 
+    private static void configureQueryResolvers(final RuntimeWiring.Builder builder) {
+        builder.type("Query", typeWiring -> typeWiring
+                .dataFetcher("search", new AuthenticatedResolver<>(
+                        new SearchResolver(SEARCHABLE_TYPES)))
+                .dataFetcher("autoComplete", new AuthenticatedResolver<>(
+                        new AutoCompleteResolver(SEARCHABLE_TYPES)))
+                .dataFetcher("browse", new AuthenticatedResolver<>(
+                        new BrowseResolver(BROWSABLE_TYPES)))
+                .dataFetcher("browsePaths", new AuthenticatedResolver<>(
+                        new BrowsePathsResolver(BROWSABLE_TYPES)))
```

Comment:
why do we need two resolvers here?

Reply:
The first retrieves a set of browse results given a _path_.

The second retrieves a set of _paths_ given an _urn_. 

They are different business logic, with different inputs and outputs. Thus, they are kept separate! But generally, if you can browse a particular type, you can fetch a browse path for an entity of the same type

Reply:
Thanks for the explanation